### ISSUE TITLE
Route Helpers in Templates can't distinguish between a path that requires a variable, and path that doesn't

### ISSUE THREAD

**Issue by L1ghtman2k** [gobuffalo/buffalo#1941]
# Description

 Route Helpers in templates can't distinguish between a path that requires a variable, and path that doesn't. This means that if I have paths: /test/ and /test/{name}, I will not be able to refer to test in a template as follows:
<%= testPath() %> because buffalo makes user specify named parameters. One way to go around that limitation is to specify the parameter as nil:
<%= testPath({name: nil}) %>, however that will print out an extra forward slash: /test//

### Steps to Reproduce the Problem

  Please describe in painful detail what you did (so others can play along with you) to get to this point. This includes things like the exact command(s) you used, or the `curl` command you used, that sort of thing.

  1. Create a custom route that takes no parameters, and a name parameter:
```
		app.GET("/test", TestHandler)
		app.GET("/test/{name}", TestHandler)
```
  2. Use default Handler:
```
func TestHandler(c buffalo.Context) error{
	return c.Render(200, r.HTML("index.html"))
}
```
  3. Let index.plush.html have <%= testPath() %> at the end of the file

### Expected Behavior

`<%= testPath() %> resolves to /test/`

### Actual Behavior

`
index.html: line 57: could not call testPath function: missing parameters for /test/{name}/: mux: missing route variable "name"
`

### Info

Please run `buffalo info` and paste the information below where it says "PASTE_HERE".

<details>

```
-> Go: Checking installation
←[32m✓ The `go` executable was found on your system at: C:\Go\bin\go.exe←[0m

-> Go: Checking minimum version requirements
←[32m✓ Your version of Go, 1.13.7, meets the minimum requirements.←[0m

-> Go: Checking Package Management
←[32m✓ You are using Go Modules (`go`) for package management.←[0m

-> Go: Checking PATH
←[31m✘ Your PATH (PATH NOT FOUND) does not contain C:\Users\aibek\go/bin.←[0m

Without C:\Users\aibek\go/bin in your `PATH` any Go executables can not be run globally.

For help setting up your Go environment please follow the instructions for you platform at:

https://www.gopherguides.com/courses/preparing-your-environment-for-go-development

-> Node: Checking installation
←[32m✓ The `node` executable was found on your system at: C:\Program Files\nodejs\node.exe←[0m

-> Node: Checking minimum version requirements
←[32m✓ Your version of Node, v12.16.1, meets the minimum requirements.←[0m

-> NPM: Checking installation
←[32m✓ The `npm` executable was found on your system at: C:\Program Files\nodejs\npm.cmd←[0m

-> NPM: Checking minimum version requirements
←[32m✓ Your version of NPM, 6.13.4, meets the minimum requirements.←[0m

-> Yarn: Checking installation
←[32m✓ The `yarnpkg` executable was found on your system at: C:\Program Files (x86)\Yarn\bin\yarnpkg.cmd←[0m

-> Yarn: Checking minimum version requirements
←[32m✓ Your version of Yarn, 1.22.4, meets the minimum requirements.←[0m

-> PostgreSQL: Checking installation
←[31m✘ The `postgres` executable could not be found on your system.←[0m
For help setting up your Postgres environment please follow the instructions for you platform at:

https://www.postgresql.org/download/

-> MySQL: Checking installation
←[32m✓ The `mysql` executable was found on your system at: C:\Program Files\MariaDB 10.4\bin\mysql.exe←[0m

-> MySQL: Checking minimum version requirements
←[32m✓ Your version of MySQL, 15.1, meets the minimum requirements.←[0m

-> SQLite3: Checking installation
←[31m✘ The `sqlite3` executable could not be found on your system.←[0m
For help setting up your SQLite3 environment please follow the instructions for you platform at:

https://www.sqlite.org/download.html

-> Cockroach: Checking installation
←[32m✓ The `cockroach` executable was found on your system at: C:\WINDOWS\system32\cockroach.exe←[0m

-> Cockroach: Checking minimum version requirements
←[32m✓ Your version of Cockroach, 19.2.5, meets the minimum requirements.←[0m

-> Buffalo (CLI): Checking installation
←[32m✓ The `buffalo` executable was found on your system at: C:\ProgramData\chocolatey\bin\buffalo.exe←[0m

-> Buffalo (CLI): Checking minimum version requirements
←[32m✓ Your version of Buffalo (CLI), v0.15.5, meets the minimum requirements.←[0m

-> Buffalo: Application Details
Pwd         C:\Users\aibek\go\src\testBuffaloApp\cokemaria
Root        C:\Users\aibek\go\src\testBuffaloApp\cokemaria
GoPath      C:\Users\aibek\go
PackagePkg  testBuffaloApp/cokemaria
ActionsPkg  testBuffaloApp/cokemaria/actions
ModelsPkg   testBuffaloApp/cokemaria/models
GriftsPkg   testBuffaloApp/cokemaria/grifts
WithModules true
Name        cokemaria
Bin         bin\cokemaria
VCS         git
WithPop     true
WithSQLite  false
WithDep     false
WithWebpack true
WithNodeJs  true
WithYarn    true
WithDocker  true
WithGrifts  true
AsWeb       true
AsAPI       false
InApp       true
PackageJSON {map[build:webpack -p --progress dev:webpack --watch]}

-> Buffalo: config/buffalo-app.toml
name = "cokemaria"
bin = "bin\\cokemaria"
vcs = "git"
with_pop = true
with_sqlite = false
with_dep = false
with_webpack = true
with_nodejs = true
with_yarn = true
with_docker = true
with_grifts = true
as_web = true
as_api = false

-> Buffalo: config/buffalo-plugins.toml
[[plugin]]
  binary = "buffalo-pop"
  go_get = "github.com/gobuffalo/buffalo-pop"

-> Buffalo: go.mod
module testBuffaloApp/cokemaria

go 1.14

require (
        github.com/cockroachdb/cockroach-go v0.0.0-20200312223839-f565e4789405 // indirect
        github.com/gobuffalo/buffalo v0.14.11
        github.com/gobuffalo/buffalo-pop v1.23.1
        github.com/gobuffalo/envy v1.9.0
        github.com/gobuffalo/fizz v1.9.8 // indirect
        github.com/gobuffalo/flect v0.2.1 // indirect
        github.com/gobuffalo/genny v0.6.0 // indirect
        github.com/gobuffalo/mw-csrf v0.0.0-20190129204204-25460a055517
        github.com/gobuffalo/mw-forcessl v0.0.0-20180802152810-73921ae7a130
        github.com/gobuffalo/mw-i18n v0.0.0-20190129204410-552713a3ebb4
        github.com/gobuffalo/mw-paramlogger v0.0.0-20190129202837-395da1998525
        github.com/gobuffalo/nulls v0.4.0 // indirect
        github.com/gobuffalo/packd v1.0.0 // indirect
        github.com/gobuffalo/packr/v2 v2.7.1
        github.com/gobuffalo/pop v4.13.1+incompatible
        github.com/gobuffalo/validate v2.0.4+incompatible // indirect
        github.com/markbates/grift v1.1.0
        github.com/pkg/errors v0.9.1 // indirect
        github.com/unrolled/secure v0.0.0-20190103195806-76e6d4e9b90c
        golang.org/x/sync v0.0.0-20200317015054-43a5402ce75a // indirect
)
```

</details>

**Comment by paganotoni** [comment]
@L1ghtman2k thanks for reporting this one. I will work on this one, meanwhile you can do something like:

```
app.GET("/test", TestHandler)
app.GET("/test/{name}", TestHandler).Alias("testNamePath")
```

To distinguish the routes.

**Comment by paganotoni** [comment]
Thanks for reporting this one @L1ghtman2k. It should be fixed in `v0.16.5`.
